mobile/src/bundles/common/components/button/button.tsx

```diff
@@ -1,19 +1,122 @@
-import React from 'react';
+import { type IconDefinition } from '@fortawesome/free-solid-svg-icons';
+import {
+    type FontAwesomeIconStyle,
+    type Props,
+} from '@fortawesome/react-native-fontawesome';
+import { FontAwesomeIcon } from '@fortawesome/react-native-fontawesome';
+import React, { useCallback, useState } from 'react';
 import { Pressable } from 'react-native';
+import {
+    type PressableProps,
+    type StyleProp,
+    type ViewStyle,
+} from 'react-native';
 
 import { Text } from '~/bundles/common/components/components';
 
 import { styles } from './styles';
 
 type Properties = {
-    label: string;
-    onPress: () => void;
-};
+    label?: string;
+    isPrimary?: boolean;
+    labelStyles?: StyleProp<ViewStyle>;
+    icon?: IconDefinition;
+    iconProps?: Omit<Props, 'icon'>;
+} & PressableProps;
+
+type StyleRecord = Record<string, unknown>;
+
+const Button: React.FC<Properties> = ({
+    label = '',
+    labelStyles,
+    style,
+    isPrimary = true,
+    disabled = false,
+    icon,
+    iconProps,
+    ...properties
```

Comment:
Let's keep it simple :) 
```suggestion
    ...props
```

Reply:
I wish, but linter said: "The variable `props` should be named `properties_`. A more descriptive name will do too.eslint[unicorn/prevent-abbreviations](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/v48.0.1/docs/rules/prevent-abbreviations.md)"

Reply:
Merge latest development and error will disappear